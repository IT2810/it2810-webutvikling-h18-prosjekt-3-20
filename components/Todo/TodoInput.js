import React, { Component } from 'react';
import {
  Text,
  View,
  StyleSheet,
  TextInput,
} from 'react-native';
import PropTypes from 'prop-types';
import { Button } from 'react-native-elements';
import Colors from '../../constants/Colors';


const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
  },
  formContainer: {
    paddingTop: 10,
    display: 'flex',
    flexDirection: 'row',
  },
  header: {
  },
  textInput: {
    backgroundColor: Colors.tabIconDefault,
    paddingLeft: 10,
    paddingRight: 10,
    height: 30,
    width: 250,
  },
  button: {
    paddingTop: 0,
    paddingBottom: 0,
    height: 30,
    backgroundColor: Colors.buttonBackground,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: Colors.buttonBorder,
  },
});

export default class TodoInput extends Component {
  constructor(props) {
    super(props);
    this.state = {
      text: '',
    };
  }

  static propTypes = {
    onTodoAdd: PropTypes.func.isRequired,
  }

  changeTextHandler = (text) => {
    this.setState({ text });
  };

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.header}>
          New TODO
        </Text>
        <View style={styles.formContainer}>
          <TextInput
            style={styles.textInput}
            onChangeText={this.changeTextHandler}
            value={this.state.text}
          />
          <Button
            buttonStyle={styles.button}
            underlayColor={Colors.tabIconDefault}
            title = {'Add'}
            onPress={() => {
              this.props.onTodoAdd({
                name: this.state.text,
                key: new Date().toString(),
              });
            }}
          />
        </View>
      </View>
    );
  }
}
