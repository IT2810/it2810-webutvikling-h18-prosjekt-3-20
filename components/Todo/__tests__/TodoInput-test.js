import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TodoInput from '../TodoInput';

describe('<TodoInput/>', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TodoInput onTodoAdd={jest.fn()}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // test tekst input
});
