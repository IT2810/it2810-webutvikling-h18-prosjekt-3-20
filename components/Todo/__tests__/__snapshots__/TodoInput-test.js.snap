// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoInput/> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={Object {}}
  >
    New TODO
  </Text>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "marginLeft": "auto",
        "marginRight": "auto",
        "paddingTop": 10,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      style={
        Object {
          "backgroundColor": "#ccc",
          "height": 30,
          "paddingLeft": 10,
          "paddingRight": 10,
          "width": 250,
        }
      }
      value=""
    />
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#337ab7",
                "borderColor": "#2e6da4",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 30,
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontSize": 17.5,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Add
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
